File: packages/client-python/sikuligo/scaffold/python/examples/find.py
from __future__ import annotations

from bootstrap_runtime import ensure_project_venv_python

ensure_project_venv_python()

from bootstrap import ensure_sikuligo_on_path
from sikuligo import Pattern, Screen

ensure_sikuligo_on_path()


def main() -> int:
    screen = Screen()
    try:
        match = screen.find(Pattern("assets/pattern.png").exact(), timeout_millis=3000)
        print(
            f"match rect=({match.x},{match.y},{match.w},{match.h}) "
            f"score={match.score:.3f} target=({match.target_x},{match.target_y})"
        )
    finally:
        screen.close()
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: packages/api/cmd/sikuligrpc/examples/node/find.mjs
import { ensureSikuligoOnPath } from "./bootstrap.mjs";
import { Screen, Pattern } from "@sikuligo/sikuligo";

ensureSikuligoOnPath();

const screen = await Screen();
try {
  const pattern = Pattern("assets/pattern.png").exact();
  const match = await screen.find(pattern);
  console.log(`found rect=(${match.x},${match.y},${match.w},${match.h}) target=(${match.targetX},${match.targetY})`);
} finally {
  await screen.close();
}
